IRTweaks/IRTweaks/Modules/Misc/DeathChanceStatFix.cs

```diff
@@ -0,0 +1,73 @@
+﻿using BattleTech;
+using Harmony;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Emit;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IRTweaks.Modules.Misc
+{
+    [HarmonyPatch(typeof(Contract), "FinalizeKilledMechWarriors")]
+    static class Contract_FinalizeKilledMechWarriors
+    {
+        public static bool Prepare() => Mod.Config.Fixes.DeathChanceStat;
+
+        public static float CalcChance(float minchance, float basechance, float gutsReduction, float gutsAmount, SimGameState s, Pilot p)
+        {
+            float statmodifier = p.LethalInjuries ? GetStat(s, "LethalDeathChance") : GetStat(s, "IncapacitatedDeathChance");
+            return Math.Max(minchance, basechance - gutsReduction * gutsAmount + statmodifier);
+        }
+
+        public static float GetStat(SimGameState s, string stat)
+        {
+            if (s.CompanyStats.ContainsStatistic(stat))
+            {
+                return s.CompanyStats.GetValue<float>(stat);
+            }
+            return 0;
+        }
+
+        public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstruction> code)
```

Comment:
I tend to find transpilers extremely hard to read, and prefer prefixes that reimplement the vanilla code (with a comment on how it differs). Not a blocker to merging, just a personal preference.

Reply:
i usually prefer a Transpiler, because it has less chance of conflicting with something else. Added a comment explaning what exactly i change, to make it more clear.